src/openforms/forms/migrations/0042_explanation_template_data_migration.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 3.2.14 on 2022-08-26 15:29
+from django.db import migrations
+
+
+def append_explanation_template_message(apps, schema_editor):
+    from openforms.authentication.registry import register as auth_register
+    from django.utils.translation import gettext_lazy as _
+
+    Form = apps.get_model("forms", "Form")
+    for form in Form.objects.all():
+        plugins = form.authentication_backends
+        login_options_count = len(
+            [plugin_id for plugin_id in plugins if plugin_id in auth_register._registry]
+        )
+        can_login = login_options_count > 0
+        form_login_required = any(
+            form_step.form_definition.login_required
+            for form_step in form.formstep_set.all()
+        )
```

Comment:
you can prefetch the form steps to prevent a cascade of queries. But even better would be to filter only the forms that require login directly in the queryset on line 10.

Reply:
Good catch. Is optimizing querysets in a migration file worth the time investment?

Reply:
Always consider the scale of the data you're dealing with. If an organization has a 1000 forms, that's 1000 additional queries that could've been avoided.

So yes, it is worth the time investment, especially if you don't know how many objects could be affected. Assume the worst.

Reply:
@dariomory Typing out this whole discussion took more time than adding a prefetch, add the prefetch please.